include/deal.II/grid/connectivity.h

```diff
@@ -0,0 +1,1477 @@
+              const auto offset_i = std::get<1>(keys[i]);
+
+              if (ref_key != std::get<0>(keys[i]))
+                {
+                  ref_key = std::get<0>(keys[i]);
+                  new_key = additional_info_compatibility[offset_i];
+                }
+
+              std::get<0>(keys[i]) = new_key;
+            }
+
+          std::sort(keys.begin(), keys.end());
+        }
+
+      std::array<unsigned int, key_length> ref_key;
+      std::array<unsigned int, key_length> ref_indices;
+      std::fill(ref_key.begin(), ref_key.end(), 0);
+
+      for (unsigned int i = 0, counter = dealii::numbers::invalid_unsigned_int;
+           i < keys.size();
+           i++)
+        {
+          const auto offset_i = std::get<1>(keys[i]);
+
+          if (ref_key != std::get<0>(keys[i]))
+            {
+              // new key
+              counter++;
+              ref_key     = std::get<0>(keys[i]);
+              ref_indices = std::get<0>(additional_info[offset_i]);
+
+              ptr_0.push_back(col_0.size());
+              for (const auto j : std::get<0>(additional_info[offset_i]))
+                if (j != 0)
+                  col_0.push_back(j - offset);
+
+              // take its orientation as default
+              col_d[offset_i]        = counter;
+              orientations[offset_i] = 1;
+            }
+          else
+            {
+              col_d[offset_i] = counter;
+              orientations[offset_i] =
+                compute_orientation(ref_indices,
+                                    std::get<0>(additional_info[offset_i]),
+                                    std::get<1>(additional_info[offset_i]));
+            }
+        }
+      ptr_0.push_back(col_0.size());
+    }
+
+
+
+    /**
+     * Call the right templated function to be able to use std::array instead
+     * of std::vector.
+     */
+    template <typename FU>
+    void
+    build_entity(const unsigned int                                d,
+                 const std::vector<std::shared_ptr<CellTypeBase>> &cell_types,
+                 const std::vector<unsigned int> &cell_types_index,
+                 const CRS<unsigned int> &        crs,
+                 CRS<unsigned int> &              crs_d,
+                 CRS<unsigned int> &              crs_0,
+                 std::vector<unsigned char> &     orientations,
+                 const FU &                       second_key_function)
+    {
+      std::size_t key_length = 0;
+
+      for (unsigned int c = 0; c < cell_types_index.size(); c++)
+        {
+          const auto &cell_type = cell_types[cell_types_index[c]];
+          for (unsigned int e = 0; e < cell_type->n_entities(d); e++)
+            key_length =
+              std::max(key_length, cell_type->vertices_of_entity(d, e).size());
+        }
+
+      // clang-format off
+    if(key_length == 2)
+      build_entity_templated<2>(d, cell_types, cell_types_index, crs, crs_d, crs_0, orientations, second_key_function);
```

Comment:
can you break the line at 80 characters, i.e., have the argument list in two lines?